packages/core/database/src/transaction-context.ts

```diff
@@ -18,8 +18,15 @@ const storage = new AsyncLocalStorage<Store>();
 
 const transactionCtx = {
   async run<TCallback extends Callback>(store: Knex.Transaction, cb: TCallback) {
+    // Check if store already exists
+    const oldStore = storage.getStore();
+
     return storage.run<ReturnType<TCallback>, void[]>(
-      { trx: store, commitCallbacks: [], rollbackCallbacks: [] },
+      {
+        trx: store,
+        commitCallbacks: oldStore?.commitCallbacks || [],
+        rollbackCallbacks: oldStore?.rollbackCallbacks || [],
+      },
```

Comment:
This fix was found by @Marc-Roig, it's because nested transactions weren't inheriting the parent transaction context. Marc will ship another dedicated PR with tests but bundling the change in this PR too allows us to move on with this PR